extensions/wikia/PortableInfobox/controllers/PortableInfoboxParserTagController.class.php

```diff
@@ -23,13 +24,16 @@ public static function parserTagInit( Parser $parser ) {
 	 * @returns String $html
 	 */
 	public function renderInfobox( $text, $params, $parser, $frame ) {
-		$connector = new InfoboxServiceConnector();
-		$renderer = new PortableInfoboxRenderService();
-		$json = $connector->getJsonBySource( '<' . self::PARSER_TAG_NAME . '>' . $text . '</' . self::PARSER_TAG_NAME . '>',
-			$frame->getNamedArguments() );
-		$data = $this->parseData( json_decode( $json ), $parser, $frame );
 
-		return $renderer->renderInfobox($data);
+		$markup = '<' . self::PARSER_TAG_NAME . '>' . $text . '</' . self::PARSER_TAG_NAME . '>';
+
+		$infoboxParser = new Wikia\PortableInfobox\Parser\Parser( $frame->getNamedArguments() );
+		$infoboxParser->setExternalParser( (new Wikia\PortableInfobox\Parser\MediaWikiParserService( $parser, $frame ) ) );
```

Comment:
external parser is required for Parser to work, maybe it should be passed in constructor